### ISSUE TITLE
Invalid regular expression flags catch on no-octal-escape rule

### ISSUE THREAD

**Issue by DanCaton** [eslint/eslint#12620]
**Tell us about your environment**

* **ESLint Version:** 6.7.1
* **Node Version:** 10.16.3
* **npm Version:** 6.12.1

**What parser (default, Babel-ESLint, etc.) are you using?**  babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": ["airbnb", "prettier"],
  "parser": "babel-eslint",
  "env": {
    "jest": true,
    "browser": true
  },
  "rules": {
    "no-restricted-syntax": ["error", "LabeledStatement", "WithStatement"],
    "no-console": "off",
    "react/prop-types": "off",
    "react/destructuring-assignment": "off",
    "react/jsx-filename-extension": "off",
    "no-param-reassign": "off",
    "spaced-comment": "off",
    "no-nested-ternary": "off",
    "no-return-assign": "off",
    "no-else-return": "off"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Linter is catching an invalid regular expressions error in the no-octal-escape rule file.
Changing the regular expression flags to `/u` seems to fix the issue.

**What did you expect to happen?**

ESLint to operate as usual

**What actually happened? Please include the actual, raw output from ESLint.**
![image](https://user-images.githubusercontent.com/56726772/69824442-afb07380-1203-11ea-8e60-6402546b2f7c.png)

**Comment by mdjermanovic** [comment]
Hi @DanCaton, thanks for the report!

This could be the same issue as #12538.

Do you see the same error when running ESLint from the command line?

**Comment by DanCaton** [comment]
Looks like it could be, I was seeing this error from Atom as well.

I don't see the same error when running ESLint from the command line this morning - though I am also not seeing the error in Atom either - no code has been changed, just a restart of my PC

**Comment by nkas17** [comment]
I am also seeing this exact error in VSCode when using the extension - https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

**Comment by nkas17** [comment]
I'm not seeing an issue from the command line

**Comment by mdjermanovic** [comment]
@nkas17 thanks for all the details! 

From your input, it seems that the VSCode extension is running ESLint in a Node version that ESLint doesn't support (`v8.9.3`):

```
[Info  - 1:43:10 PM] ESLint server running in node v8.9.3
```

Could you please open an issue in the extension's repository:

https://github.com/Microsoft/vscode-eslint

**Comment by nkas17** [comment]
yes, I saw that too and thought that was odd,  I'll post a question over there, thank you!

**Comment by nkas17** [comment]
My issues were due to running an old VSCode version which ran under Node v8.9.3.  Once I updated to the latest VSCode version which runs under Node v12.4.0 the Eslint extension ran under v12.4.0 too and no I longer encountered this issue.

**Comment by kaicataldo** [comment]
Closing this as it looks like this question has been answered.

**Comment by trevjnels** [comment]
Howdy - know this has been closed - but if anyone is experiencing this error in prettier-atom a simple update to atom fixed it. I was running 1.3.something
